### PR TITLE
Add preliminary support for logging in via app role

### PR DESCRIPTION
**Pull request by jchambers**
This deeeeeeefinitely doesn't cover the full scope of the [AppRole auth backend](https://www.vaultproject.io/docs/auth/approle.html), but it is enough to log in with an existing role and secret.

### CODE REVIEW

File: src/main/java/com/bettercloud/vault/api/Auth.java [hunk 1]
```diff
@@ -203,6 +203,67 @@ public AuthResponse loginByAppID(final String path, final String appId, final St
...
+     *
+     * <blockquote>
+     * <pre>{@code
+     * final AuthResponse response = vault.auth().loginByAppRole("approle", "app_id", "user_id");
+     *
+     * final String token = response.getAuthClientToken());
+     * }</pre>
+     * </blockquote>
+     *
+     * @param path The path on which the authentication is performed (e.g. <code>auth/app-id/login</code>)
```
Comment (MyRealityCoding) [inline src/main/java/com/bettercloud/vault/api/Auth.java]: 'approle' instead of 'app-id'

File: src/main/java/com/bettercloud/vault/api/Auth.java [hunk 2]
```diff
@@ -203,6 +203,67 @@ public AuthResponse loginByAppID(final String path, final String appId, final St
     }
 
     /**
+     * <p>Basic login operation to authenticate to an app-role backend.  Example usage:</p>
+     *
+     * <blockquote>
+     * <pre>{@code
+     * final AuthResponse response = vault.auth().loginByAppRole("approle", "app_id", "user_id");
+     *
+     * final String token = response.getAuthClientToken());
```
Comment (MyRealityCoding) [inline src/main/java/com/bettercloud/vault/api/Auth.java]: Remove one `)` character to have valid Java code.

File: src/main/java/com/bettercloud/vault/api/Auth.java (hunk 1 above)
Comment (jchambers) [inline src/main/java/com/bettercloud/vault/api/Auth.java]: Done.

File: src/main/java/com/bettercloud/vault/api/Auth.java (hunk 2 above)
Comment (jchambers) [inline src/main/java/com/bettercloud/vault/api/Auth.java]: Done.

**Comment by jchambers** [comment]
@bitbrain I completely missed your comments back in the day; beg your pardon! Everything should be resolved now.

**Comment by steve-perkins** [comment]
Thanks so much for the PR, and apologies for the lengthy delay in getting it merged!

I'm made some further changes in the integration test suite, to programmatically fetch a `roleId` and `secretId` when the test suite starts up.  So devs won't have to do those steps manually, and set JVM system properties for them.

**Pull request merged**